Makefile

```diff
@@ -4,8 +4,14 @@ PHONY += all clean install uninstall gen_nv_symvers
 .PHONY: $(PHONY)
 
 KVER := $(shell uname -r)
+OFA_ARCH := $(shell uname -m)
 OFA_DIR ?= /usr/src/ofa_kernel
-OFA_KERNEL ?= $(shell ( test -d $(OFA_DIR)/$(KVER) && echo $(OFA_DIR)/$(KVER) ) || ( test -d $(OFA_DIR)/default && echo $(OFA_DIR)/default ) || ( test -d /var/lib/dkms/mlnx-ofed-kernel/ && ls -d /var/lib/dkms/mlnx-ofed-kernel/*/build ) || ( echo $(OFA_DIR) ))
+OFA_CANDIDATES = $(OFA_DIR)/$(OFA_ARCH)/$(KVER) $(OFA_DIR)/$(KVER) $(OFA_DIR)/default /var/lib/dkms/mlnx-ofed-kernel
+OFA_KERNEL ?= $(shell for d in $(OFA_CANDIDATES); do if [ -d "$$d" ]; then echo "$$d"; exit 0; fi; done; echo $(OFA_DIR))
```

Comment:
what do you need the extra echo $(OFA_DIR) for?

Reply:
I left the existing code as is. This is a shell script snippet inside $(shell ), that is supposed to print a path. It defaults to print $(OFA_DIR) if nothing else was found and I left that default as is.

I'm not sure it is a good default, but changing it would be left for a future changeset.